Use patchable package refrences

Rosuavio opened the pull request:
This should make it so that dependents can patch this package's dependencies. It also specifies on what version's this package depends on.

dfordivam committed:
Use try_trait_v2, fixes build with rust 1.61

dfordivam committed:
add try_option

Rosuavio committed:
Use patchable package refrences

Rosuavio commented:
For packages that are not on crates.io, dependents would also have to specify a patch that can find the package. From my understanding that need would be alleviated with a private cargo registry. @jonored I think you know more than me on private cargo registries, so you might be able to talk more on that.

Ericson2314 commented on Cargo.toml:
```diff
@@ -6,11 +6,16 @@ edition = "2018"
-nanos_sdk = { git = "https://github.com/LedgerHQ/ledger-nanos-sdk.git" }
-nanos_ui = { git = "https://github.com/LedgerHQ/ledger-nanos-ui.git" }
-ledger-log = { git = "https://github.com/alamgu/ledger-log.git" }
+nanos_sdk = "0.1.0"
```
See if you can do this and patch?
```suggestion
nanos_sdk = { version = "*", registry = "alamgu" }
```

Ericson2314 commented on Cargo.toml:
```diff
@@ -6,11 +6,16 @@ edition = "2018"
 [features]
 speculos = ["nanos_sdk/speculos"]
+
+[patch.crates-io]
```
```suggestion
[patch.alamgu]
```

Rosuavio committed:
Use any version nanos_sdk, nanos_ui, ledger-log

These libs are all pre-release and will be updating rapidly.
We should have no version constraint for now.

Rosuavio commented on .cargo/config.toml:
```diff
@@ -0,0 +1,2 @@
+[registries]
+alamgu = { index = "https://github.com/alamgu/TODO" }
```
@Ericson2314 @jonored, without us having a registry repo existing, I put this for now. Without it would not build at all, even though it does not check the registry because of our patches.

Specifying a registry can happen other ways https://doc.rust-lang.org/cargo/reference/registries.html

Also don't know if we NEED to do this right now per say given that this repo does not build on its own anyway.

jonored replied:
Seems legit to me.

Rosuavio replied:
I must have missed it, it does actually check the registry is there. Yeah using this wont work until we have a registry setup, should I just do the registry stuff later?

Ericson2314 replied:
@Rosuavio I like that plan.

Rosuavio replied:
Done, https://github.com/alamgu/ledger-crypto-helpers/tree/ref-registry

jonored committed:
Updates to be a bit more sane and use the new SDK.

Note: This is not guaranteed to zeroize secrets correctly, due to
returning ECPrivateKey types in a few places; need to resolve.

jonored committed:
Compress eddsa public keys.

Ericson2314 committed:
Merge branch 'main' into dn-support-rust-1.61

Ericson2314 committed:
Merge branch 'main' into dn-support-rust-1.61

Ericson2314 committed:
Bump version number

Ericson2314 committed:
Merge branch 'main' into dn-support-rust-1.61

Ericson2314 committed:
Merge branch 'main' into jg-update-for-new-sdk

Ericson2314 committed:
Merge branch 'dn-support-rust-1.61' into jg-update-for-new-sdk

Ericson2314 committed:
Improve change log

Ericson2314 committed:
Merge pull request #3 from alamgu/jg-update-for-new-sdk

Update for the new SDK

Ericson2314 committed:
Generalize the error type of `with_public_keys`

dfordivam committed:
Add Write impl for Blake2b

dfordivam committed:
add "*_int" APIs which work on ECPrivateKey

dfordivam committed:
Remove unused import

Ericson2314 committed:
Merge pull request #4 from alamgu/dn-write-impl-and-int-apis

Add Write impl and _int APIs

Ericson2314 committed:
Update hasher.rs

Ericson2314 committed:
Fix ed25519 module bug; make with_public_keys easier to use

Ericson2314 committed:
Merge remote-tracking branch 'origin/develop' into with_public_keys-generalize-error

Ericson2314 committed:
Merge pull request #6 from alamgu/with_public_keys-generalize-error

Changes for Pocket

Ericson2314 committed:
Merge branch 'develop' into patchable

dfordivam committed:
Manually encode to base64, avoiding 1kb stack allocation

jonored committed:
Merge remote-tracking branch 'origin/dn-hash-stack-fix' into hasher-dispatch

Ericson2314 committed:
Apply suggestions from code review

jonored committed:
Decouple hash formatting choice from hashers.

jonored committed:
Make hash length an associated const for Hasher.

Ericson2314 committed:
Merge branch 'develop' into dn-hash-stack-fix

Ericson2314 committed:
Merge branch 'dn-hash-stack-fix' into hasher-dispatch

Ericson2314 committed:
Fix accidental infinite recursion

Ericson2314 committed:
Add some Default instances

Ericson2314 committed:
Move instance to be closer to type

It was in an odd spot.

Ericson2314 committed:
Fix bound on `Hash` for `Zeroizing` impl

Ericson2314 committed:
Get rid of `HasBufSize` trait

It is not needed.

Ericson2314 committed:
Merge pull request #8 from alamgu/hasher-dispatch

Decouple the hash formatting choice from hash algorithm

Ericson2314 committed:
Merge branch 'develop' into patchable
